custom_components/smartthinq_sensors/wideq/range.py

```diff
@@ -0,0 +1,207 @@
+    DeviceStatus,
+    UNITTEMPMODES,
+    UNIT_TEMP_FAHRENHEIT,
+    UNIT_TEMP_CELSIUS,
+)
+
+
+_LOGGER = logging.getLogger(__name__)
+
+OVEN_TEMP_UNIT = {
+    "0": UNITTEMPMODES.Fahrenheit,
+    "1": UNITTEMPMODES.Celsius,
+}
+
+
+class RangeDevice(Device):
+    """A higher-level interface for a cooking range."""
+
+    def __init__(self, client, device):
+        super().__init__(client, device, RangeStatus(self, None))
+
+    def reset_status(self):
+        self._status = RangeStatus(self, None)
+        return self._status
+
+    def poll(self) -> Optional["RangeStatus"]:
+        """Poll the device's current state."""
+
+        res = self.device_poll("ovenState")
+        if not res:
+            return None
+
+        self._status = RangeStatus(self, res)
+        return self._status
+
+
+class RangeStatus(DeviceStatus):
+    """Higher-level information about an range's current status.
+
+    :param device: The Device instance.
+    :param data: JSON data from the API.
+    """
+
+    def __init__(self, device, data):
+        super().__init__(device, data)
+        self._oven_temp_unit = None
+
+    def _update_features(self):
+        result = [
+            self.cooktop_left_front_state,
+            self.cooktop_left_rear_state,
+            self.cooktop_center_state,
+            self.cooktop_right_front_state,
+            self.cooktop_right_rear_state,
+            self.oven_lower_state,
+            self.oven_lower_target_temp,
+            self.oven_upper_state,
+            self.oven_upper_target_temp,
+            self.oven_temp_unit,
+        ]
+        return
+
+    def _get_oven_temp_unit(self):
+        if not self._oven_temp_unit:
+            oven_temp_unit = self.lookup_enum(["MonTempUnit"])
+            if not oven_temp_unit:
+                self._oven_temp_unit = STATE_OPTIONITEM_NONE
+            else:
+                self._oven_temp_unit = (
+                    OVEN_TEMP_UNIT.get(oven_temp_unit, UNITTEMPMODES.Celsius)
+                ).value
+        return self._oven_temp_unit
+
+    @property
+    def is_on(self):
+        return self.is_cooktop_on or self.is_oven_on
+
+    @property
+    def is_cooktop_on(self):
+        result = [
+            self.cooktop_left_front_state,
+            self.cooktop_left_rear_state,
+            self.cooktop_center_state,
+            self.cooktop_right_front_state,
+            self.cooktop_right_rear_state,
+        ]
+        for r in result:
+            if r != 'Off':
+                return True
+        return False
+        
+    @property
+    def cooktop_left_front_state(self):
+        """For some cooktops (maybe depending on firmware or model), the
+        five burners do not report individual status. Instead, the 
+        cooktop_left_front reports aggregated status for all burners.
+        """
+        status = self.lookup_enum(["LFState"])
+        return self._update_feature(
+            FEAT_COOKTOP_LEFT_FRONT_STATE, status, True
+        )
```

Comment:
I think that all this checks should be changed a little bit to not be language dependent:

- Import from `device` the constant `STATE_OPTIONITEM_OFF`
- define a constant for `enum` that identify `off` status (eg: `OVEN_OFF_STATUS = "@OV_STATE_INITIAL_W"`)
- change the code here in this way:

```
status = self.lookup_enum("LFState")
if status and status == OVEN_OFF_STATUS:
   status = STATE_OPTIONITEM_OFF
return self._update_feature(
   FEAT_COOKTOP_LEFT_FRONT_STATE, status
)
```

Reply:
Done.